solr/core/src/java/org/apache/solr/cloud/ZkDistributedQueue.java

```diff
@@ -51,9 +51,16 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * A ZK-based distributed queue. Optimized for single-consumer,
+ * <p>A ZK-based distributed queue. Optimized for single-consumer,
  * multiple-producer: if there are multiple consumers on the same ZK queue,
- * the results should be correct but inefficient
+ * the results should be correct but inefficient.</p>
+ *
+ * <p>This implementation (with help from subclass {@link OverseerTaskQueue}) is used for the
+ * <code>/overseer/collection-queue-work</code> queue used for Collection and Config Set API calls to the Overseer.</p>
+ *
+ * <p>In order to enqueue a message into this queue, a {@link CreateMode#EPHEMERAL_SEQUENTIAL} response node is created
```

Comment:
I think we can cordon this off behind `@implNote` and `@implSpec` Javadocs tags to improve the readability

Reply:
Thanks, didn't know about these tags.

Reply:
I'm not the only one not to know about these tags: ant precommit doesn't know either...